arches/app/models/models.py

```diff
@@ -520,16 +520,20 @@ def is_editable(self):
         else:
             return True
 
-    def get_published_graph(self, language=None, raise_if_missing=False):
+    def get_published_graph(self, language=None):
         if not language:
             language = translation.get_language()
 
         try:
             graph = PublishedGraph.objects.get(publication=self.publication, language=language)
         except PublishedGraph.DoesNotExist:
-            if raise_if_missing:
-                raise
-            graph = None
+            try:
+                from arches.app.utils.i18n import LanguageSynchronizer  # prevents circular import
+
+                LanguageSynchronizer.synchronize_settings_with_db()
```

Comment:
I retested #10079, and when using the language switcher, the page just hangs with a spinner (no response is received from Django). This was with dockerized AfS.

I added a print statement before the call to `synchronize...` , including the graph name, and got:

```py
2024-03-05 14:18:22 syncing... 1 Instrument
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:22 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Arches System Settings
2024-03-05 14:18:23 syncing... 1 Attribute Assignment 
2024-03-05 14:18:23 syncing... 1 Name
2024-03-05 14:18:23 syncing... 1 Dimension
2024-03-05 14:18:23 syncing... 1 Monetary Amount
2024-03-05 14:18:23 syncing... 1 TimeSpan
2024-03-05 14:18:24 syncing... 1 __Activity_Template
2024-03-05 14:18:24 syncing... 1 Statement
2024-03-05 14:18:24 syncing... 1 Identifier
2024-03-05 14:18:24 syncing... 1 Instrument
2024-03-05 14:18:24 syncing... 1 Collection or Set
2024-03-05 14:18:24 syncing... 1 Project
2024-03-05 14:18:24 syncing... 1 Observation
2024-03-05 14:18:25 syncing... 1 Physical Thing
2024-03-05 14:18:25 syncing... 1 Person
2024-03-05 14:18:26 syncing... 1 Place
2024-03-05 14:18:26 syncing... 1 Textual Work
2024-03-05 14:18:26 syncing... 1 Sampling Activity
2024-03-05 14:18:26 syncing... 1 Modification
2024-03-05 14:18:27 syncing... 1 Digital Resources
```

Reply:
Looks like pg was choking:
```
2024-03-05 14:25:41 2024-03-05 19:25:41.146 PST [1706] LOG:  unexpected EOF on client connection with an open transaction
```